### PR TITLE
Removes two unused AnalysisConfig options

### PR DESCRIPTION
**Pull request by benwtrent**
`result_finalization_window` is pretty much unused, and over time has become broken due to other features. Consequently, it and `overlapping_buckets` are being removed as options for AnalysisConfig.

Since these were undocumented and largely unfinished/broken options, removal is acceptable and helps simplify much of the downstream ml-cpp code. 

Relates to https://github.com/elastic/ml-cpp/issues/302

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/AnalysisConfig.java [hunk 1]
```diff
@@ -130,14 +121,6 @@ public TimeValue getBucketSpan() {
         return bucketSpan;
     }
 
-    public String getCategorizationFieldName() {
```
Comment (droberts195) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/AnalysisConfig.java]: This is an accessor for a different field - was it removed in error?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/AnalysisConfig.java [hunk 2]
```diff
@@ -130,14 +121,6 @@ public TimeValue getBucketSpan() {
...
-        return categorizationFieldName;
-    }
-
-    public List<String> getCategorizationFilters() {
```
Comment (droberts195) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/AnalysisConfig.java]: This is an accessor for a different field - was it removed in error?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java [hunk 3]
```diff
@@ -146,8 +135,13 @@ public AnalysisConfig(StreamInput in) throws IOException {
         summaryCountFieldName = in.readOptionalString();
         detectors = Collections.unmodifiableList(in.readList(Detector::new));
         influencers = Collections.unmodifiableList(in.readList(StreamInput::readString));
-        overlappingBuckets = in.readOptionalBoolean();
-        resultFinalizationWindow = in.readOptionalLong();
+
+        // BWC for result_finalization_window and overlapping_buckets
+        // TODO Remove in 7.0.0
+        if (in.getVersion().before(Version.V_6_6_0)) {
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java]: This is going to make the CI for this PR fail. It's actually quite complex how to deal with this without breaking the build. I'll reach out to you to explain.

Comment (benwtrent) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java]: Yeah, I noticed some previous pains you experience when handling the the multiple bucket span options...Would like to learn the best path forward :D.

**Review by droberts195 — COMMENTED** [review]
Looks good apart from a couple of questionable removals

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/AnalysisConfig.java (hunk 1 above)
Comment (benwtrent) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/AnalysisConfig.java]: Definitely in error.

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**